crates/nu-parser/src/parser.rs

```diff
@@ -3684,19 +3697,60 @@ pub fn parse_list_expression(
 
             if let LiteElement::Command(_, command) = arg {
                 while spans_idx < command.parts.len() {
-                    let arg = parse_multispan_value(
-                        working_set,
-                        &command.parts,
-                        &mut spans_idx,
-                        element_shape,
-                    );
+                    let curr_span = command.parts[spans_idx];
+                    let curr_tok = working_set.get_span_contents(curr_span);
+                    let (arg, ty) = if curr_tok == b"..." {
+                        // Parse the spread operator
+                        spans_idx += 1;
+                        if spans_idx < command.parts.len() {
+                            let spread_arg = parse_multispan_value(
+                                working_set,
+                                &command.parts,
+                                &mut spans_idx,
+                                &SyntaxShape::OneOf(vec![
+                                    SyntaxShape::List(Box::new(element_shape.clone())),
+                                    SyntaxShape::String,
+                                ]),
+                            );
+                            let elem_ty = match &spread_arg.ty {
+                                Type::List(elem_ty) => *elem_ty.clone(),
+                                Type::String => Type::String,
+                                _ => Type::Any,
```

Comment:
Not sure this case is reachable, since the syntax shape is restricted to lists and strings.